src/libraries/Microsoft.Extensions.Configuration.Abstractions/src/ConfigurationPath.cs

```diff
@@ -65,7 +67,7 @@ public static string GetSectionKey(string path)
         /// </summary>
         /// <param name="path">The path.</param>
         /// <returns>The original path minus the last individual segment found in it. Null if the original path corresponds to a top level node.</returns>
-        public static string GetParentPath(string path)
+        public static string? GetParentPath(string? path)
```

Comment:
honestly I think `path` should be marked as `non-null` the doc suggests it should be a path also it's not intuitive what will happen when it's null. I do not feel strongly about this so fine with me if you think we should keep it as is

Reply:
I also don't have a strong opinion. One benefit of making `path` nullable is that user don't need additional null-check in order to call this method (if their variable may be null).

I didn't find any calls to this method here or in aspnet, so don't know how this method it used.